src/Stratis.Bitcoin.Features.Miner/PosBlockAssembler.cs

```diff
@@ -21,27 +21,25 @@ public class PosBlockAssembler : PowBlockAssembler
 
         public PosBlockAssembler(
             IConsensusLoop consensusLoop,
-            Network network,
-            MempoolSchedulerLock mempoolLock,
-            ITxMempool mempool,
             IDateTimeProvider dateTimeProvider,
-            IStakeChain stakeChain,
-            IStakeValidator stakeValidator,
-            ChainedBlock chainTip,
             ILoggerFactory loggerFactory,
-            AssemblerOptions options = null)
-            : base(chainTip, consensusLoop, dateTimeProvider, loggerFactory, mempool, mempoolLock, network, options)
+            ITxMempool mempool,
+            MempoolSchedulerLock mempoolLock,
+            Network network,
+            IStakeChain stakeChain,
+            IStakeValidator stakeValidator)
+            : base(consensusLoop, dateTimeProvider, loggerFactory, mempool, mempoolLock, network, new AssemblerOptions() { IsProofOfStake = true })
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.stakeChain = stakeChain;
             this.stakeValidator = stakeValidator;
         }
 
-        public override BlockTemplate CreateNewBlock(Script scriptPubKeyIn, bool fMineWitnessTx = true)
+        public override BlockTemplate Build(ChainedBlock chainTip, Script minerAddress)
         {
-            this.logger.LogTrace("({0}.{1}:{2},{3}:{4})", nameof(scriptPubKeyIn), nameof(scriptPubKeyIn.Length), scriptPubKeyIn.Length, nameof(fMineWitnessTx), fMineWitnessTx);
+            this.logger.LogTrace("({0}:{1},{2}.{3}:{4})", nameof(chainTip), chainTip.Height, nameof(minerAddress), nameof(minerAddress.Length), minerAddress.Length);
```

Comment:
When you log chained block, just use `{0}:'{1}'` and `nameof(chainTip), chainTip`, not just its height

Reply:
Fixed 👍